File: SDKs/github.com/ghes-3.0/1.1.4/python/src/sdk/models/operations/orgs_get_webhook_config_for_org.py
from dataclasses import dataclass, field
from typing import Optional
from sdk.models import shared


@dataclass
class OrgsGetWebhookConfigForOrgPathParams:
    hook_id: int = field(metadata={'path_param': { 'field_name': 'hook_id', 'style': 'simple', 'explode': False }})
    org: str = field(metadata={'path_param': { 'field_name': 'org', 'style': 'simple', 'explode': False }})
    

@dataclass
class OrgsGetWebhookConfigForOrgRequest:
    path_params: OrgsGetWebhookConfigForOrgPathParams = field()
    

@dataclass
class OrgsGetWebhookConfigForOrgResponse:
    content_type: str = field()
    status_code: int = field()
    webhook_config: Optional[shared.WebhookConfig] = field(default=None)
    

File: SDKs/github.com/ghes-3.1/1.1.4/typescript/src/sdk/models/operations/orgsgetwebhookconfigfororg.ts
import { Metadata, SpeakeasyBase } from "../../../internal/utils/utils";
import * as shared from "../shared";


export class OrgsGetWebhookConfigForOrgPathParams extends SpeakeasyBase {
  @Metadata({ data: "pathParam, style=simple;explode=false;name=hook_id" })
  hookId: number;

  @Metadata({ data: "pathParam, style=simple;explode=false;name=org" })
  org: string;
}


export class OrgsGetWebhookConfigForOrgRequest extends SpeakeasyBase {
  @Metadata()
  pathParams: OrgsGetWebhookConfigForOrgPathParams;
}


export class OrgsGetWebhookConfigForOrgResponse extends SpeakeasyBase {
  @Metadata()
  contentType: string;

  @Metadata()
  statusCode: number;

  @Metadata()
  webhookConfig?: shared.WebhookConfig;
}
